Made test warnings fatal

Azad99-9 opened the pull request:
<!--
This section can be deleted after reading.

We employ the following branching strategy to simplify the development process and to ensure that only stable code is pushed to the `master` branch:

- `develop`: For unstable code: New features and bug fixes.
- `master`: Where the stable production ready code lies. Only security related bugs.

NOTE!!!

ONLY SUBMIT PRS AGAINST OUR `DEVELOP` BRANCH. THE DEFAULT IS `MAIN`, SO YOU WILL HAVE TO MODIFY THIS BEFORE SUBMITTING YOUR PR FOR REVIEW. PRS MADE AGAINST `MAIN` WILL BE CLOSED.
-->

<!--
Thanks for submitting a pull request! Please provide enough information so that others can review your pull request.
-->

**What kind of change does this PR introduce?**

This PR makes `flutter test warnings` fatal.

**Issue Number:**

Fixes #1467 

**Did you add tests for your changes?**

Yes

**Snapshots/Videos:**

<!--Add snapshots or videos wherever possible.-->

**If relevant, did you update the documentation?**

<!--Add link to Talawa-Docs.-->

**Summary**

- Added a new file `test/flutter_test_config.dart`.
- This file gets executed `prior` to every test file in test folder, so that devs can configure test parameters at a single place.
- Added  `WidgetController.hitTestWarningShouldBeFatal = true` which makes all hitTest warnings fatal.
- Fixed all the `existing warnings` in the tests.

**Does this PR introduce a breaking change?**

No

**Other information**

<!--Add extra information about this PR here-->

**Have you read the [contributing guide](https://github.com/PalisadoesFoundation/talawa/blob/master/CONTRIBUTING.md)?**

Yes

Azad99-9 commented:
@Cioppolo14 please assign reviewers for this PR.

literalEval commented on lib/views/after_auth_screens/events/explore_event_dialogue.dart:
```diff
@@ -22,153 +19,135 @@ class _ExploreEventDialogState extends State<ExploreEventDialog> {
-              Column(
-                crossAxisAlignment: CrossAxisAlignment.start,
+      insetPadding:
+          EdgeInsets.symmetric(horizontal: SizeConfig.screenWidth! * 0.15),
```
Why remove the trailing comma ?

literalEval commented on lib/views/pre_auth_screens/set_url.dart:
```diff
@@ -37,189 +37,193 @@ class _SetUrlState extends State<SetUrl> {
+                                size: Size(
+                                  SizeConfig.screenWidth! * 0.125,
+                                  SizeConfig.screenWidth! * 0.125 * 0.5,
+                                ), //You can Replace [WIDTH] with your desired width for Custom Paint and height will be calculated automatically
```
Remove these comments as part of this PR itself. They serve no purpose.

literalEval reviewed (changes requested):
1. Are the extra files modified so that their tests can be changed too ?
2. Sometimes the warning comes from an old package or about something which we can't immediately fix. How to proceed in that case ?

Azad99-9 commented:
@literalEval 

> Are the extra files modified so that their tests can be changed too ?

I didn't get you kindly elaborate.

> Sometimes the warning comes from an old package or about something which we can't immediately fix. How to proceed in that case ?

- At the moment all the warnings previously existed were related to `hitTests` so I have added `WidgetController.hitTestWarningShouldBeFatal = true`. Additionally I have added a new file `flutter_test_config` which can be configured for all the warnings that are supported by `flutter testing framework`. 
- For the warnings which are `unforeseen` or `can't be fixed immediately` we can define a GitHub action However this won't work when we run test in CLI using `flutter test` which is also a requirement in the issue this PR is targetting. I am open to any solution from your side that covers this unexpected situation.

Azad99-9 commented:
@literalEval 

> I meant that you have changed some files and their tests too. It is so because these files were generating warnings earlier ?

- Yes, these files were generating warnings earlier because their UI wasn't `responsive`.
- As the `MediaQuery` constraints are `tight` in `test environment` this caused some UI elements to `exceed the screen` and when we try to `tap` these overflowed widgets the `hitTest` warnings were being generated.
- So I have made the repective UI components responsive while `preserving the existing functionality` and modified the tests to adhere to these new changes.

Azad99-9 commented:
> > Are the extra files modified so that their tests can be changed too ?
> 
> I meant that you have changed some files and their tests too. It is so because these files were generating warnings earlier ?
> 
> About the 'not instantly fixable warnings, I think if we fix the existing warnings for now, we may afford to add new features with full coverage ? I mean, say if we install a new package that doesn't have full test support, or a warning in itself, then we might choose to install another package that does the same job ? Though all of the major and popular packages have 100% coverage, there's always a culprit. And that's the concern here.

I think choosing an alternative in such situations would be a reasonable solution and I agree with this. coming to the existing warnings they are fixed in this PR without altering the existing functionality.

Azad99-9 commented:
@literalEval one instance of such UI components mentioned in [this](https://github.com/PalisadoesFoundation/talawa/pull/2332#issuecomment-1899857776) comment. Can be seen [here](https://github.com/PalisadoesFoundation/talawa/issues/2336#issue-2090310740).

Azad99-9 commented:
@literalEval kindly review.

literalEval reviewed (approved):
LGTM
